### PR TITLE
SLACK: improve fetching pyspark profiles and clean up getImports for …

### PR DESCRIPTION
**Pull request by jaebchoi**
…Pyspark Step

### CODE REVIEW

File: foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/generator/TargetedPipelineStepPythonGenerator.java [hunk 1]
```diff
@@ -38,10 +38,10 @@ public void generate(GenerationContext generationContext) {
         for (Step step : pythonTargetPipeline.getSteps()) {
             if (shouldGenerateStep(step, generationContext)) {
                 VelocityContext vc = getNewVelocityContext(generationContext);
-                String profileName = MavenUtil.getPySparkDataDeliveryProfileName(generationContext);
+                boolean isRecordGenerationInPipelineModule = MavenUtil.isRecordGenerationInPipelineModule(generationContext);
                 PythonStep pythonStep = new PythonStep(step);
                 pythonStep.validate();
-                pythonStep.setProfileName(profileName);
+                pythonStep.setRecordGenerationInPipelineModule(isRecordGenerationInPipelineModule);
```
Comment (carter-cundiff) [inline foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/generator/TargetedPipelineStepPythonGenerator.java]: Could simplify this to one line: `pythonStep.setRecordGenerationInPipelineModule(MavenUtil.isRecordGenerationInPipelineModule(generationContext));`

Comment (jaebchoi) [reply foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/generator/TargetedPipelineStepPythonGenerator.java]: sounds good, updated!

File: foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/metamodel/element/python/PythonStep.java [hunk 2]
```diff
@@ -140,12 +140,13 @@ private Set<String> getImports(boolean isImplModule) {
...
-                if(profileName.equals("data-delivery-pyspark"))
+                if(isRecordGenerationInPipelineModule)
                 {
                     importsSet.add(moduleImport.replace("from ",
                             isImplModule ? "from .." : "from ..."));
-                }else if(profileName.equals("data-delivery-pyspark-pipeline")){
-                    String rootArtifact = rootArtifactId.replace("-", "_");
+                }else{
+                    String rootArtifact = PipelineUtils.deriveLowerSnakeCaseNameFromHyphenatedString(rootArtifactId);
+                    //String rootArtifact = rootArtifactId.replace("-", "_");
```
Comment (carter-cundiff) [inline foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/metamodel/element/python/PythonStep.java]: Removed old commented out code

**Review by carter-cundiff — APPROVED** [review]

File: foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/generator/util/MavenUtil.java [hunk 3]
```diff
@@ -221,7 +222,7 @@ public String getPackaging() {
      * @param context GenerationContext
      * @return the profile for pyspark data delivery pipeline
      */
-    public static String getPySparkDataDeliveryProfileName(GenerationContext context) {
+    public static boolean isRecordGenerationInPipelineModule(GenerationContext context) {
```
Comment (cwoods-cpointe) [inline foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/generator/util/MavenUtil.java:225]: a: The java docs should be updated to reflect the change in return value and function expectation

**Pull request merged**